### PR TITLE
feat(router): add return_amount in response data

### PR DESCRIPTION
**Pull request by epanchee**
1. Cherry-pick router contract from Astroport [v3.2.2](https://github.com/astroport-fi/astroport-core/tree/v3.3.2)
2. Set response data with return amount. It is processed in reply after all swaps via router finished.
3. Increase router code coverage.

### CODE REVIEW

File: contracts/router/src/contract.rs [hunk 1]
```diff
@@ -234,7 +234,9 @@ fn assert_minimum_receive(
             amount: swap_amount,
         })
     } else {
-        Ok(Response::default())
+        Ok(Response::default().set_data(to_binary(&SwapResponseData {
+            return_amount: swap_amount,
+        })?))
```
Comment (gmanev7) [inline contracts/router/src/contract.rs:239]: Not a good idea to set it only if `minimum_receive` is set. In general, the response should be set in all successful executions.

**Review by gmanev7 — COMMENTED** [review]
Would suggest replacing the logic [here ](https://github.com/astroport-fi/astroport-core/blob/3b7c0c5681f6b287e61cadc8045431e11d2fbae0/contracts/router/src/contract.rs#L181-L197) with a `reply` entry point. If the pair [swap execution](https://github.com/astroport-fi/astroport-core/blob/3b7c0c5681f6b287e61cadc8045431e11d2fbae0/contracts/pair/src/contract.rs#L758-L777) returns the amount then the router would not re-evaluate the `return_amount`.

File: contracts/router/src/contract.rs (hunk 1 above)
Comment (epanchee) [reply contracts/router/src/contract.rs:239]: Certainly. Makes sense

**Comment by donovansolms** [comment]
LGTM. @gmanev7 could you please confirm that [the data returned](https://github.com/astroport-fi/astroport-core/pull/386/files#diff-8370b0ba8af4a2a69cb972239dd4aca40f4ad4f13a21ef1af1f9cec9d25556c5R71) and the rest of this PR provides everything you need on your end?

**Comment by gmanev7** [comment]
> LGTM. @gmanev7 could you please confirm that [the data returned](https://github.com/astroport-fi/astroport-core/pull/386/files#diff-8370b0ba8af4a2a69cb972239dd4aca40f4ad4f13a21ef1af1f9cec9d25556c5R71) and the rest of this PR provides everything you need on your end?

In terms of the data we expect - yes. In terms of the correct execution - we're going to test.

**Review by donovansolms — APPROVED** [review]

**Pull request merged**